### ISSUE TITLE
Running files that include uppercase letter generates warning

### ISSUE THREAD

**Issue by pheki** [fornwall/rust-script#19]
Running rust-script in any file that includes uppercase letter generates a crate with the same name, which cargo 
complains.

While rust files are usually modules and as such should be lowercase, sometimes the user may choose a name that matches his project better. Also [cargo-make uses temporary files that may contain uppercase letters](https://github.com/sagiegurari/cargo-make/issues/551).

My proposed solution would be to convert the file name to snake case when using as the crate name. `serde` can do this for struct fields, so I suppose a simple solution already exists in the ecosystem. I can give it a try if you would accept.

Script.rs:

```rust
fn main() {
    println!("test");
}
```

Output:

```console
$ rust-script Script.rs 
warning: crate `Script_5920885be5fa777047cfed70` should have a snake case name
  |
  = note: `#[warn(non_snake_case)]` on by default
  = help: convert the identifier to snake case: `script_5920885be5fa777047cfed70`

warning: 1 warning emitted

test
```

**Comment by fornwall** [comment]
> My proposed solution would be to convert the file name to snake case when using as the crate name. serde can do this for struct fields, so I suppose a simple solution already exists in the ecosystem. I can give it a try if you would accept.

Agreed that we should definitely get rid of the warning in case of upper case in file names!

How about just lower casing the file name (to keep it simple and avoid pulling in a snake casing dependency)?

**Comment by pheki** [comment]
> How about just lower casing the file name (to keep it simple and avoid pulling in a snake casing dependency)?

Yes! That's a much better solution, as the crate name itself doesn't really matter... I'll work on it

**Comment by fornwall** [comment]
Thanks! This has now been released in version 0.14.0.
